### PR TITLE
Preferences page redesign

### PR DESCRIPTION
**Pull request by aleksfront**
* Cleaning whole Preferences view
* Spreading content onto separate tabs
* UI tweaks and fixes
* Introducing new `<Switcher/>` component

Preferences with dark theme

https://user-images.githubusercontent.com/9607060/113563881-238d3800-9611-11eb-8e98-304bfd224399.mp4

Preferences with light theme

https://user-images.githubusercontent.com/9607060/113563905-2a1baf80-9611-11eb-8bec-652c1fdaf0a5.mp4

### CODE REVIEW

File: extensions/survey/renderer.tsx [hunk 1]
```diff
@@ -8,6 +8,7 @@ export default class SurveyRendererExtension extends LensRendererExtension {
   appPreferences = [
     {
       title: "In-App Surveys",
+      showInPreferencesTab: "telemetry",
```
Comment (aleksfront) [inline extensions/survey/renderer.tsx:11]: Telemetry extensions are meant to be moved in separate tab away from other ones.

File: src/renderer/components/+preferences/helm-charts.tsx [hunk 2]
```diff
@@ -106,6 +104,7 @@ export class HelmCharts extends React.Component {
             formatOptionLabel={this.formatOptionLabel}
             controlShouldRenderValue={false}
             className="box grow"
+            themeName="lens"
```
Comment (aleksfront) [inline src/renderer/components/+preferences/helm-charts.tsx:107]: Created another theme for `<Select/>` to preserve it until all UI components will have unifying look. Now inputs in "settings" pages look different (intentionally) from same elements in the dashboard.

File: src/renderer/components/+preferences/preferences.tsx [hunk 3]
```diff
@@ -59,28 +106,58 @@ export class Preferences extends React.Component {
...
               <Select
                 options={this.themeOptions}
                 value={preferences.colorTheme}
                 onChange={({ value }: SelectOption) => preferences.colorTheme = value}
+                themeName="lens"
               />
             </section>
-            <section id="proxy">
+
+            <hr className="small"/>
```
Comment (aleksfront) [inline src/renderer/components/+preferences/preferences.tsx:128]: `.small` hrs and sections are having smaller margins.

File: src/renderer/components/layout/page-layout.scss [hunk 4]
```diff
@@ -35,96 +27,172 @@
...
+          width: 35px;
+          height: 35px;
+          display: grid;
+          place-items: center;
+          border: 2px solid var(--textColorDimmed);
+          border-radius: 50%;
+          cursor: pointer;
+
+          &:hover {
+            background-color: #72767d4d;
```
Comment (aleksfront) [inline src/renderer/components/layout/page-layout.scss:119]: This color keeps unchanged regardless of the active theme.

**Comment by miskun** [comment]
Visually, looking good!

**Review by jakolehm — APPROVED** [review]
LGTM

File: src/renderer/components/+preferences/preferences.tsx [hunk 5]
```diff
@@ -45,9 +55,46 @@ export class Preferences extends React.Component {
...
+  renderNavigation() {
+    const extensions = appPreferenceRegistry.getItems().filter(e => !e.showInPreferencesTab);
+
+    return (
+      <Tabs className="flex column" scrollable={false} onChange={this.onTabChange} value={this.activeTab}>
+        <div className="header">Preferences</div>
+        <Tab value={Pages.Application} label="Application" data-testid="application-tab"/>
+        <Tab value={Pages.Proxy} label="Proxy" data-testid="proxy-tab"/>
+        <Tab value={Pages.Kubernetes} label="Kubernetes" data-testid="kube-tab"/>
+        <Tab value={Pages.Telemetry} label="Telemetry" data-testid="telemetry-tab"/>
```
Comment (ixrock) [inline src/renderer/components/+preferences/preferences.tsx:71]: Probably better to have a list of `{pageId: string, label: string, testId?: string, content: React.ReactNode}` and render them via `[].map()`?

File: src/renderer/components/switch/form-switcher.tsx [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    "& .MuiTypography-root": {
+      fontSize: 14,
+      fontWeight: 500,
+      flex: 1,
+      color: "var(--textColorAccent)"
+    }
+  },
+});
+
+export function FormSwitch(props: FormControlLabelProps) {
```
Comment (ixrock) [inline src/renderer/components/switch/form-switcher.tsx:17]: What it's for? How does it work?

**Review by ixrock — APPROVED** [review]

File: src/renderer/components/switch/form-switcher.tsx (hunk 6 above)
Comment (aleksfront) [reply src/renderer/components/switch/form-switcher.tsx:17]: `FormSwitch` wraps `Switcher` component and places some label close to it. It allows to do such a layout:
FormSwitch

File: src/renderer/components/+preferences/preferences.tsx (hunk 5 above)
Comment (aleksfront) [reply src/renderer/components/+preferences/preferences.tsx:71]: Yes, will change it on next iteration 👍🏻

File: src/renderer/components/+preferences/helm-charts.tsx [hunk 7]
```diff
@@ -116,20 +115,18 @@ export class HelmCharts extends React.Component {
...
           {Array.from(this.addedRepos).map(([name, repo]) => {
-            const tooltipId = `message-${name}`;
-
             return (
-              <Badge key={name} className="added-repo flex gaps align-center justify-space-between">
-                <span id={tooltipId} className="repo">{name}</span>
+              <div key={name} className="repo flex gaps align-center justify-space-between">
+                <div>
+                  <div className="repoName">{name}</div>
+                  <div className="repoUrl">{repo.url}</div>
```
Comment (Nokel81) [inline src/renderer/components/+preferences/helm-charts.tsx:122]: Are you sure that the URL won't be too long in general?

Comment (aleksfront) [reply src/renderer/components/+preferences/helm-charts.tsx:122]: They will wrap text content onto new lines, so nothing bad happened in this case.

**Pull request merged**